### DISCUSSION TITLE
Add `forge-clone`

### DISCUSSION THREAD

**Discussion by sellout** [magit/forge#547] [Ideas]
A convenience wrapper around `magit-clone` that
1. accepts forge URLs (“https://github.com/sellout/forge”) in addition to repo URLs (“git@github.com:sellout/forge.git”),
2. sets up remotes based on where the repo has been forked from (`git remote add upstream git@github.com:magit/forge.git`),
3. executes `forge-pull` (maybe configurable to do `forge-add-repository` instead), and
4. maybe also provide a setting for automatically rebinding `magit-clone` to `forge-clone`.

Do (or should) other wrappers like this exist?

**Comment by tarsius** [comment]
1. `magit-clone` already supports that and always did.
2. Don't name a remote `upstream`; that's what `origin` is for. See https://docs.magit.vc/magit/Branching.html#The-Two-Remotes-1.
3. `forge-add-repository` now has more options, so we should likely always present that, instead of going straight for `forge-pull`. At which point we have the issue that the clone is created asynchronously, then at some point a menu to configure forge pulling pops up. Which isn't that different from first invoking `magit-clone` and then later (also manually) `forge-add-repository`.
4. snip

I don't think that gives us much and...

(Personally I plan to move away from forges such as Github. I was planning to move to Codeberg, but am not sure anymore, that is enough. Developments over the last few and especially the last year, forced me to invest in defense, and I am increasingly worried that having a world-writable public todo list puts maintainers under constant stress. I'll keep maintaining Forge; after all most (i.e., currently all) of my packages are still hosted on Github and I also want to keep using it for repositories on Github, which are owned by other people. But new features are becoming increasingly unlikely.)
